cluster-up/cluster/k3d/common.sh

```diff
@@ -2,312 +2,173 @@
-    _wait_containers_ready
-    _run_registry "$KIND_DEFAULT_NETWORK"
-
-    for node in $(_get_nodes | awk '{print $1}'); do
-        _configure_registry_on_node "$node" "$KIND_DEFAULT_NETWORK"
-        _configure_network "$node"
-    done
-    prepare_config
+function _install_k3d() {
+    echo "STEP: Install k3d"
+    curl -s https://raw.githubusercontent.com/k3d-io/k3d/main/install.sh | TAG=$K3D_TAG bash
 }
 
-function _add_worker_extra_mounts() {
-  cat <<EOF >> ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-  extraMounts:
-  - containerPath: /var/log/audit
-    hostPath: /var/log/audit
-    readOnly: true
-EOF
-
-    if [[ "$KUBEVIRT_PROVIDER" =~ sriov.* || "$KUBEVIRT_PROVIDER" =~ vgpu.* ]]; then
-        cat <<EOF >> ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-  - containerPath: /dev/vfio/
-    hostPath: /dev/vfio/
-EOF
-  fi
+function _extract_kubeconfig() {
+    echo "STEP: Extract kubeconfig"
+    k3d kubeconfig print $CLUSTER_NAME > ${BASE_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig
 }
 
-function _add_worker_kubeadm_config_patch() {
-    cat << EOF >> ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-  kubeadmConfigPatches:
-  - |-
-    kind: JoinConfiguration
-    nodeRegistration:
-      kubeletExtraArgs:
-        "feature-gates": "CPUManager=true"
-        "cpu-manager-policy": "static"
-        "kube-reserved": "cpu=500m"
-        "system-reserved": "cpu=500m"
-EOF
-}
-
-function _add_workers() {
-    # appending eventual workers to the yaml
-    for ((n=0;n<$(($KUBEVIRT_NUM_NODES-1));n++)); do
-        cat << EOF >> ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-- role: worker
-EOF
-    _add_worker_kubeadm_config_patch
-    _add_worker_extra_mounts
+function _label_agents() {
+    echo "STEP: label agents"
+    for node in $(_get_nodes); do
+        if [[ "$node" =~ .*"agent".* ]]; then
+            _kubectl label node $node node-role.kubernetes.io/worker=worker
+        fi
     done
 }
 
-function _add_kubeadm_config_patches() {
-    if [ $KUBEVIRT_WITH_KIND_ETCD_IN_MEMORY == "true" ]; then
-        cat <<EOF >> ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-kubeadmConfigPatches:
-- |
-  kind: ClusterConfiguration
-  metadata:
-    name: config
-  etcd:
-    local:
-      dataDir: $ETCD_IN_MEMORY_DATA_DIR
-EOF
-        echo "KIND cluster etcd data will be mounted to RAM on kind nodes: $ETCD_IN_MEMORY_DATA_DIR"
-    fi
-}
+function _create_cluser() {
+    echo "STEP: Create cluster"
 
-function _prepare_kind_config() {
-    _add_workers
-    _add_kubeadm_config_patches
+    id1=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-1
+    id2=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-2
+    id3=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-3
+    printf '%.0s1' {1..32} > ${id1}
+    printf '%.0s2' {1..32} > ${id2}
+    printf '%.0s3' {1..32} > ${id3}
 
-    echo "Final KIND config:"
-    cat ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-}
+    NETWORK=bridge
+    if [ $CRI_BIN == podman ]; then
```

Comment:
I do not fully understand.
If it is podman it pulls the image and if it is not, it does not load it?

Reply:
This is the temporary WA for running podman locally
when we are using docker, docker supports short-names, i.e `fedora` which stands for docker.io/fedora
podman doesnt support it
the main server image that k3d pulls is shortended, so podman cant pull it, so we help it by that, and then all works (locally)

Reply:
Removing it, with podman 4 it works

For fedora this can be used
https://podman.io/blogs/2022/03/06/why_no_podman4_f35.html